Wabbajack.Lib/AInstaller.cs

```diff
@@ -279,12 +279,41 @@ public async Task<bool> DownloadArchive(Archive archive, bool download, Absolute
         public async Task HashArchives()
         {
             Utils.Log("Looking for files to hash");
-            var toHash = DownloadFolder.EnumerateFiles()
+            // Enumerate all files.
+            var allFiles = DownloadFolder.EnumerateFiles()
                 .Concat(Game.GameLocation().EnumerateFiles())
                 .ToList();
+
+            // Create a Dictionary with filesize as the key.
+            var hashDict = new Dictionary<long, List<AbsolutePath>>();
+
+            // Populate the Dictionary with all files we found, using a list for the paths to allow for duplicated filesizes.
+            allFiles.ForEach(o =>
+            {
+                if (hashDict.ContainsKey(o.Size))
+                {
+                    hashDict[o.Size].Add(o);
+                }
+                else
+                {
+                    hashDict.Add(o.Size, new List<AbsolutePath> { o });
+                }
+            });
+
```

Comment:
This can just be:

```csharp
var hashDict = allFiles.GroupBy(f => f.Size).ToDictionary(group => group.Key);
```

Reply:
Man, I really need to learn all the things Linq allows, this massively simplifies it, thanks.